### DISCUSSION TITLE
Slice-level-control

### DISCUSSION THREAD

**Discussion by abida-02** [srsran/oran-sc-ric#42] [General]
In the srsran (5G) side i am able to connect 3 cots ue with two different  slice. Now from the oran-sc-ric provided xapp, i want to use the xapp to adjust the prb of the each slice in the RAN side. In the e2sm_rc_module.py, the sst and sd are not exposed as parameter. If i want to expose this parameter, how can i enable it in the xapp side?

**Comment by pgawlowicz** [comment]
currently, the parameter is not used on the gnb side, so setting it on the xapp side changes nothing.

**Reply by abida-02** [reply]
I enabled the sd and sst parameter in the gnb side. Now i want to use these parameter in the xapp side. Is it possible to enable this in the e2sm_rc_module.py ?

**Reply by pgawlowicz** [reply]
change here:
https://github.com/srsran/oran-sc-ric/blob/main/xApps/python/lib/e2sm_rc_module.py#L44

**Comment by abida-02** [comment]
I enabled the slice information in the e2sm_rc_module.py. I connected two ue with two different slice. Is it possible to gather the ue information from the xapp side? For example number of connected ue or slice can be collected by xapp?

**Reply by pgawlowicz** [reply]
You can use [kpm_mon_xapp.py](https://github.com/srsran/oran-sc-ric/blob/main/xApps/python/kpm_mon_xapp.py#L77) and subscribe for any available metric with Report Style 4. The gnb E2 agent will send RIC Indications with the requested metric for all connected UEs. Hence, you can derive the number of connected UEs. Unfortunately, for far, there is no filtering based on the slice IDs.
